hack/local-cmo.sh

```diff
@@ -1,28 +1,103 @@
 #!/usr/bin/env bash
+set -eu -o pipefail
+
 # Script will:
 # 1. Patch CVO to exclude CMO from being managed
 # 2. Scale in-cluster CMO to 0
 # 3. Try to run CMO locally (CMO needs to be built before starting)
 
-set -eu
 
-# Go to the root of the repo
-cd "$(git rev-parse --show-cdup)"
+validate() {
+  local ret=0
+
+  [[ -x operator ]] || {
+    echo "Cannot find local operator binary. Did you run 'make build' ?"
```

Comment:
Yeah we can do that and if compilation fails and let user deal with the failure.

Reply:
There are at least 3 Makefile targets which would result in creating a binary:
- `make build` - simple way
- `make operator` - invoked by `make build` and it will try regenerating `assets/`
- `make operator-no-deps` - it will only build operator and not regenerate `assets/`

Since we have those 3 it would be better to just say that binary is missing and not running build for the user.

Reply:
done :)

Reply:
@paulfantom I am not sure if the three different make targets change things that much. `make build` will be run only if the operator binary does not exist, so the user can still build a custom one. Also, the message tells the user to run `make build` without indicating that there is some other way to build the binary.

Reply:
@paulfantom, IMHO, it makes sense to ensure that the operator runs fine and runs with the latest changes. Thus, having `make build` would reduce the edit, run, debug to just 
1. edit the code
2. run the script.

Power users are free to add make changes to script. We could even add a `NO_BUILD=true ./hack/local-cmo.sh`

Reply:
Bear in mind there are cases during development where regenerating everything in `assets/` can happen after running `make build` and when this isn't something that is desired. This isn't an uncommon pattern. On top of that this script is meant mostly for developers behind CMO and we all here are power users. Considering those I don't see any benefit in complicating script to add `NO_BUILD` feature over running two commands.
If we really want to have something that ties `make build` with this script, then let's do it in Makefile by introducing another target.

Reply:
@paulfantom makes sense. thanks for the explanation. 

I have changed the behaviour of the script back to how it was. It only checks if the `operator` binary exists. 

I have also added a `make run-local` target that `build`s and then runs the `local-cmo.sh` script.